pkg/operator/k8sutil/pod.go

```diff
@@ -186,6 +188,32 @@ func GetPodPhaseMap(pods *v1.PodList) map[v1.PodPhase][]string {
 	return podPhaseMap
 }
 
+// GetJobLog gets the logs for the pod. If there is more than one pod with the label selector, the logs from
+// the first pod will be returned.
+func GetPodLog(clientset kubernetes.Interface, namespace string, labelSelector string) (string, error) {
+	opts := metav1.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	pods, err := clientset.CoreV1().Pods(namespace).List(opts)
+	if err != nil {
+		return "", fmt.Errorf("failed to get version pod. %+v", err)
+	}
+	for _, pod := range pods.Items {
+		req := clientset.CoreV1().Pods(namespace).GetLogs(pod.Name, &v1.PodLogOptions{})
+		readCloser, err := req.Stream()
+		if err != nil {
+			return "", fmt.Errorf("failed to read from stream. %+v", err)
+		}
+
+		builder := &strings.Builder{}
+		defer readCloser.Close()
+		_, err = io.Copy(builder, readCloser)
+		return builder.String(), err
+	}
+
+	return "this is a fake version", nil
```

Comment:
I does not seem good to call a function "GetPodLog" and return output like this targeted for the Ceph specific function. This could cause potential issues from my view.

Reply:
@galexrt are you commenting on this specific return value or a general statement about the approach? This specific string needs to be changed, I was debugging with this. Instead I should be returning an error since we couldn't get the pod log to detect the version.

Reply:
Yes, I'm talking about the return value.